lib-multisrc/heancms/src/eu/kanade/tachiyomi/multisrc/heancms/HeanCms.kt

```diff
@@ -70,6 +76,36 @@ abstract class HeanCms(
         .add("Origin", baseUrl)
         .add("Referer", "$baseUrl/")
 
+    private val authHeaders: Headers by lazy {
+        val builder = headersBuilder()
+        if (preferences.loginPref && enableLogin) {
+            val token = getToken()
+            if (token != null) {
+                builder.add("Authorization", "Bearer $token")
+            }
+        }
+        builder.build()
+    }
+
+    private fun getToken(): String? {
```

Comment:
instead of logging in everytime, maybe login once and save the token in the sharedpreference along with expiration date. and if the token is near expiration, refresh it by logging in again